src/ILCompiler.Compiler/src/Compiler/DependencyAnalysis/GenericCompositionNode.cs

```diff
@@ -0,0 +1,183 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Text;
+
+using Internal.TypeSystem;
+
+using Debug = System.Diagnostics.Debug;
+
+namespace ILCompiler.DependencyAnalysis
+{
+    /// <summary>
+    /// Describes how a generic type instance is composed - the number of generic arguments, their types,
+    /// and variance information.
+    /// </summary>
+    public class GenericCompositionNode : ObjectNode, ISymbolNode
```

Comment:
This looks like in multi-module we could have sharing of these.  So you can add an override for `ShouldShareNodeAcrossModules` that returns true.